install_ffmpeg.sh

```diff
@@ -1,231 +1,3 @@
-else
-  echo "No CUDA SDK detected, building without GPU support"
-fi
-
-if [[ $BUILD_TAGS == *"debug-video"* ]]; then
-  echo "video debug mode, building ffmpeg with tools, debug info and additional capabilities for running tests"
-  DEV_FFMPEG_FLAGS="--enable-muxer=md5,flv --enable-demuxer=hls --enable-filter=ssim,tinterlace --enable-encoder=wrapped_avframe,pcm_s16le "
-  DEV_FFMPEG_FLAGS+="--enable-shared --enable-debug=3 --disable-stripping --disable-optimizations --enable-encoder=libx265,libvpx_vp8,libvpx_vp9 "
-  DEV_FFMPEG_FLAGS+="--enable-decoder=hevc,libvpx_vp8,libvpx_vp9 --enable-libx265 --enable-libvpx --enable-bsf=noise "
-else
-  # disable all unnecessary features for production build
-  DISABLE_FFMPEG_COMPONENTS+=" --disable-doc --disable-sdl2 --disable-iconv --disable-muxers --disable-demuxers --disable-parsers --disable-protocols "
-  DISABLE_FFMPEG_COMPONENTS+=" --disable-encoders --disable-decoders --disable-filters --disable-bsfs --disable-postproc --disable-lzma "
-fi
-
-if [[ ! -e "$ROOT/ffmpeg/libavcodec/libavcodec.a" ]]; then
-  git clone https://github.com/livepeer/FFmpeg.git "$ROOT/ffmpeg" || echo "FFmpeg dir already exists"
-  cd "$ROOT/ffmpeg"
-  git checkout 2e18d069668c143f3c251067abd25389e411d022
-  ./configure ${TARGET_OS:-} $DISABLE_FFMPEG_COMPONENTS --fatal-warnings \
-    --enable-libx264 --enable-gpl \
-    --enable-protocol=rtmp,file,pipe \
-    --enable-muxer=mpegts,hls,segment,mp4,hevc,matroska,webm,null --enable-demuxer=flv,mpegts,mp4,mov,webm,matroska \
-    --enable-bsf=h264_mp4toannexb,aac_adtstoasc,h264_metadata,h264_redundant_pps,hevc_mp4toannexb,extract_extradata \
-    --enable-parser=aac,aac_latm,h264,hevc,vp8,vp9 \
-    --enable-filter=abuffer,buffer,abuffersink,buffersink,afifo,fifo,aformat,format \
-    --enable-filter=aresample,asetnsamples,fps,scale,hwdownload,select,livepeer_dnn,signature \
-    --enable-encoder=aac,opus,libx264 \
-    --enable-decoder=aac,opus,h264 \
-    --extra-cflags="${EXTRA_CFLAGS} -I${ROOT}/compiled/include -I/usr/local/cuda/include" \
-    --extra-ldflags="${EXTRA_FFMPEG_LDFLAGS} -L${ROOT}/compiled/lib -L/usr/local/cuda/lib64" \
-    --prefix="$ROOT/compiled" \
-    $EXTRA_FFMPEG_FLAGS \
-    $DEV_FFMPEG_FLAGS || (tail -100 ${ROOT}/ffmpeg/ffbuild/config.log && exit 1)
-    # If configure fails, then print the last 100 log lines for debugging and exit.
-fi
-
-if [[ ! -e "$ROOT/ffmpeg/libavcodec/libavcodec.a" || $BUILD_TAGS == *"debug-video"* ]]; then
-  cd "$ROOT/ffmpeg"
-  make -j$NPROC
-  make -j$NPROC install
-fi
+echo 'WARNING: downloading and executing lpms/install_ffmpeg.sh, use it directly in case of issues'
+curl https://raw.githubusercontent.com/livepeer/lpms/d9c78b62effdb4f5c8fc438b6033da1090d04a03/install_ffmpeg.sh | bash -s $1
```

Comment:
This seems to be related to the specific commit, why not to use the `master`?

Reply:
FYI I ve been using this version on my Orchestrator for 5 days now and had no issues.

Reply:
> This seems to be related to the specific commit, why not to use the master? 

Reproducibility - the same reason tools like go mod are pinned to specific versions. Ideally we would be able to check out any go-livepeer commit and get the exact version of ffmpeg that commit depends on.

Note that we only have to update this if the LPMS install_ffmpeg.sh script changes. If LPMS is updated without any changes to `install_ffmpeg.sh`, then we don't have to actually change anything.

We could have a script or some tooling to detect mismatches (eg, must match the latest commit to `lpms/install_ffmpeg.sh` at the specific LPMS checkout which this version of go-livepeer depends on) but that can be done later if mismatches become an issue. Even better would be a more suitable build tool that is able to handle third-party dependencies, eg bazel, but that is a completely different issue.